app/services/course_search_service.rb

```diff
@@ -24,6 +28,14 @@ def call
 
 private
 
+  def should_sort_by_provider_ascending?
```

Comment:
I don't think we need the `should_` prefix on these two methods. Also, could they go under the `attr_reader`? Might just be me but I think we usually have the `attr_reader` at the top of the private bit. You could also add a reader for `:sort`.